clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -270,14 +270,7 @@ public void run() {
         while (!forceClose && transactionManager != null && transactionManager.hasOngoingTransaction()) {
             if (!transactionManager.isCompleting()) {
                 log.info("Aborting incomplete transaction due to shutdown");
-
-                try {
-                    // It is possible for the transaction manager to throw errors when aborting. Catch these
-                    // so as not to interfere with the rest of the shutdown logic.
-                    transactionManager.beginAbort();
-                } catch (Exception e) {
-                    log.error("Error in kafka producer I/O thread while aborting transaction: ", e);
-                }
+                transactionManager.beginAbort();
```

Comment:
I wonder if we could keep the try catch block, but rather than just log the error, set forceClose to true
cc: @kirktrue what do you think?

Reply:
Updated.